src/NzbDrone.Core/Indexers/IndexerBase.cs

```diff
@@ -146,6 +147,24 @@ protected virtual IList<ReleaseInfo> CleanupReleases(IEnumerable<ReleaseInfo> re
             return result.DistinctBy(v => v.Guid).ToList();
         }
 
+        protected IEnumerable<ReleaseInfo> FilterReleasesByQuery(IEnumerable<ReleaseInfo> releases, SearchCriteriaBase searchCriteria)
+        {
+            var commonWords = new[] { "a", "an", "the" };
+
+            if (!searchCriteria.IsIdSearch)
+            {
+                var splitRegex = new Regex("[^a-zA-Z0-9]+");
+
+                // split search term to individual terms for less aggressive filtering, filter common terms
+                var terms = splitRegex.Split(searchCriteria.SearchTerm).Where(t => !commonWords.Contains(t) && t.IsNotNullOrWhiteSpace() && t.Length > 1);
+
+                // check in title and description for any term searched for
+                releases = releases.Where(r => terms.Any(t => r.Title.Contains(t) || r.Description.Contains(t))).ToList();
```

Comment:
Should we look at additional release properties for the terms. Ex. user searches by imdb id but doesn't use the imdb field. By comparing term to `release.imdb` or `release.tmdb` we can possible catch those from being filtered out.

Reply:
Right agree, however this is an `or` logic, so if anything matches it will accept the release (title, description, etc). Adding the ids would just give one more possible match criteria, and not necessarily cause a match to be removed. Meaning fewer valid releases may get filtered (less false positives).